src/scale_impls/encode.rs

```diff
@@ -512,40 +492,13 @@ mod test {
             bits![0, 1, 1, 0, 0, 1],
         );
         // a single value should still encode OK:
-        assert_can_encode_to_type(Value::unnamed_composite(vec![Value::bool(false)]), bits![0]);
+        assert_can_encode_to_type(value!((false)), bits![0]);
         assert_can_encode_to_type(
-            Value::unnamed_composite(vec![
-                Value::bool(false),
-                Value::bool(true),
-                Value::bool(true),
-                Value::bool(false),
-                Value::bool(false),
-                Value::bool(true),
-            ]),
-            bits![0, 1, 1, 0, 0, 1],
-        );
-        assert_can_encode_to_type(
-            Value::unnamed_composite(vec![
-                Value::u128(0),
-                Value::u128(1),
-                Value::u128(1),
-                Value::u128(0),
-                Value::u128(0),
-                Value::u128(1),
-            ]),
-            bits![0, 1, 1, 0, 0, 1],
-        );
-        assert_can_encode_to_type(
-            Value::unnamed_composite(vec![
-                Value::i128(0),
-                Value::i128(1),
-                Value::i128(1),
-                Value::i128(0),
-                Value::i128(0),
-                Value::i128(1),
-            ]),
+            value!((false, true, true, false, false, true)),
             bits![0, 1, 1, 0, 0, 1],
         );
+        assert_can_encode_to_type(value!((0u8, 1u8, 1u8, 0u8, 0u8, 1u8)), bits![0, 1, 1, 0, 0, 1]);
+        assert_can_encode_to_type(value!((0, 1, 1, 0, 0, 1)), bits![0, 1, 1, 0, 0, 1]);
```

Comment:
Ah I guess the types aren't necessary then; that's cool! Don't think we need them anywhere then :)

Reply:
In this place they are necessary. The difference between line 500 and line 501 is, that in the first case we create an unnamed composite with U128 primitives and in the second case, they are I128 primitives. The test wants to check, that conversion from either unsigned and signed integers into bit sequences works here, I think.